### DISCUSSION TITLE
PIC32 and Trice

### DISCUSSION THREAD

**Discussion by elfux** [rokath/trice#459] [Q&A]
Hello, I'm having trouble using PIC32MZ with the xc32-gcc compiler with trice.

I got it to work with deferred output into the UART and I'm starting to try out the testCheck.c examples.

I'm having plenty of issues, such as:

` break; case __LINE__: TRice16( iD(6385), "att: 0x8888 == %04xh\n", 0x8888 ); //exp: time: 842,150_450default: att: 0x8888 == 8888h`
`break; case __LINE__: Trice16( iD(1138), "att: 0x8888 == %04xh\n", 0x8888 ); //exp: time:       5_654default: att: 0x8888 == 8888h`
`break; case __LINE__: trice16( iD(3239), "att: 0x8888 == %04xh\n", 0x8888 ); //exp: time:            default: att: 0x8888 == 8888h `

1st line prints out "att: 0x8888 == ffffh", while the others are okay.

`TCOBSv1: 03 cd c3 48 6d 38 cb 02 88 7f 94 4c 01 cc 02 88 88 a8 46 4d cd 02 88 88 a6 00`
`->TRICE: 03 cd c3 48 6d 38 cb 02 ff ff ff ff 7f 94 4c 01 cc 02 88 88 46 4d cd 02 88 88`
`COM13:       triceExamples.c    79  946,686_147  0x8888 == ffffh`

Checking memories on MPLAB seem to have the correct message on the UART:
![image](https://github.com/rokath/trice/assets/88747118/b689c1c6-9a8d-4797-aabb-47912d3a04cb)

Also, if I use "TRice" for a 32 bit timestamp, I can't seem to pass on a 32 bit int variable, run time generated.
Strings doesn't seem to be working either. 

I'm guessing it's something on my configurations that are wrong?
[triceConfig.zip](https://github.com/rokath/trice/files/15010169/triceConfig.zip)

Any help is truly appreciated, thanks!

**Comment by elfux** [comment]
Found out my error. Thanks

**Comment by rokath** [comment]
Would be really nice if you share your mistake, if it could help others.
